### PR TITLE
[Feature] Support import automatic partition creation syntax

### LINKED ISSUE #11444: Improve partition management
## Enhancement
Many databases support the function of importing and automatically creating partitions. We can support this function to complete the creation of partitions, and then use ttl to control the deletion of partitions. This ease of use will be better than the previous method.

### PR DESCRIPTION
**Pull request by Astralidea**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #11444

## Problem Summary(Required) ：

example:
```
CREATE TABLE site_access(
    event_day DATE,
    site_id INT DEFAULT '10',
    city_code VARCHAR(100),
    user_name VARCHAR(32) DEFAULT '',
    pv BIGINT DEFAULT '0'
)
DUPLICATE KEY(event_day, site_id, city_code, user_name)
PARTITION BY day(event_day) (
START ("2015-06-01") END ("2022-12-01") EVERY (INTERVAL 1 month)
)
DISTRIBUTED BY HASH(event_day, site_id) BUCKETS 32
```
This is a series of PRs, UT will join when this feature is turned on.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 1]
```diff
@@ -528,10 +528,40 @@ public ParseNode visitCreateTableStatement(StarRocksParser.CreateTableStatementC
                                 .stream().map(Identifier::getValue).collect(toList()));
     }
 
+    @Override
+    public ParseNode visitPartitionExpression(StarRocksParser.PartitionExpressionContext context) {
+        Expr expr = (Expr) visit(context.expression());
+        List<Expr> exprs = ImmutableList.of(expr);
+        if (context.DAY() != null) {
+            return new FunctionCallExpr("day", exprs);
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:536]:
```suggestion
            return new FunctionCallExpr(FunctionSet.YEAR, exprs);
```

Comment (ABingHuang) [reply fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:536]: convert to date_trunc?

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 2]
```diff
@@ -528,10 +528,40 @@ public ParseNode visitCreateTableStatement(StarRocksParser.CreateTableStatementC
...
-        final List<Identifier> identifierList = visit(context.identifierList().identifier(), Identifier.class);
-        final List<String> columnList = identifierList.stream().map(Identifier::getValue).collect(toList());
         List<PartitionDesc> partitionDescList = new ArrayList<>();
+        if (context.partitionExpression() != null) {
+            for (StarRocksParser.RangePartitionDescContext rangePartitionDescContext : context.rangePartitionDesc()) {
+                final PartitionDesc rangePartitionDesc = (PartitionDesc) visit(rangePartitionDescContext);
+                partitionDescList.add(rangePartitionDesc);
+            }
+            Expr expr = (Expr) visit(context.partitionExpression().expression());
+            if (expr instanceof SlotRef) {
```
Comment (ABingHuang) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:555]: what if expr is not SlotRef?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 28 (14.29%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|23|13.04%|[533, 534, 535, 536, 537, 538, 539, 540, 541, 542, 544, 550, 551, 552, 553, 554, 555, 556, 557, 558]|
|:large_blue_circle:|com/starrocks/sql/ast/ExpressionPartitionDesc.java|1|5|20.00%|[24, 25, 26, 27]|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 30 (13.33%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|25|12.00%|[533, 534, 535, 536, 537, 538, 539, 540, 541, 542, 543, 544, 545, 547, 553, 554, 555, 556, 557, 558, 559, 560]|
|:large_blue_circle:|com/starrocks/sql/ast/ExpressionPartitionDesc.java|1|5|20.00%|[24, 25, 26, 27]|

File: fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java [hunk 3]
```diff
@@ -528,10 +528,40 @@ public ParseNode visitCreateTableStatement(StarRocksParser.CreateTableStatementC
...
-        final List<Identifier> identifierList = visit(context.identifierList().identifier(), Identifier.class);
-        final List<String> columnList = identifierList.stream().map(Identifier::getValue).collect(toList());
         List<PartitionDesc> partitionDescList = new ArrayList<>();
+        if (context.partitionExpression() != null) {
+            for (StarRocksParser.RangePartitionDescContext rangePartitionDescContext : context.rangePartitionDesc()) {
+                final PartitionDesc rangePartitionDesc = (PartitionDesc) visit(rangePartitionDescContext);
+                partitionDescList.add(rangePartitionDesc);
+            }
+            Expr expr = (Expr) visit(context.partitionExpression().expression());
+            List<String> columnList = ImmutableList.of(((SlotRef) expr).getColumnName());
```
Comment (HangyuanLiu) [inline fe/fe-core/src/main/java/com/starrocks/sql/parser/AstBuilder.java:558]: expr may be expression?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 4 / 34 (11.76%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|3|29|10.34%|[533, 534, 535, 536, 537, 538, 539, 540, 541, 542, 543, 544, 545, 547, 553, 554, 555, 556, 557, 559, 560, 561, 562, 563, 565, 566]|
|:large_blue_circle:|com/starrocks/sql/ast/ExpressionPartitionDesc.java|1|5|20.00%|[24, 25, 26, 27]|

**Review by HangyuanLiu — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**